reporting.md

```diff
@@ -0,0 +1,615 @@
+
+Because such "downgrade attacks" are not necessarily apparent to the receiving
+MTA, this document defines a mechanism for sending domains to report on
+failures at multiple parts of the MTA-to-MTA conversation.
+
+Specifically, this document defines a reporting schema that covers failures 
+in routing, STARTTLS negotiation, and both DANE [@!RFC7671] and STS (TODO: Add
+ref) policy validation errors.
+
+Implementers establish a policy by publishing a TXT record in the DNS which
+instructs comliant sending MTAs to deliver reports of delivery and STARTTLS
+failures in the appropriate format to the specified endpoint.
+
+This document is intended as a companion to the specification for SMTP MTA 
+Strict Transport Security (MTA-STS, TODO: Add ref).
+
+## Terminology
+
+The keywords **MUST**, **MUST NOT**, **REQUIRED**, **SHALL**, **SHALL NOT**,
+**SHOULD**, **SHOULD NOT**, **RECOMMENDED**, **MAY**, and **OPTIONAL**, when
+they appear in this document, are to be interpreted as described in [@!RFC2119].
+
+We also define the following terms for further use in this document:
+
+* STS Policy: A definition of the expected TLS availability and behavior, as
+  well as the desired actions for a given domain when a sending MTA encounters
+  different results.
+* Policy Domain: The domain against which an STS Policy is defined.
+* Sending MTA: The MTA initiating the delivery of an email message.
+
+# Related Technologies
+
+  * This document is intended as a companion to the specification for SMTP MTA
+    Strict Transport Security (MTA-STS, TODO: Add ref).
+
+  * The Public Key Pinning Extension for HTTP [@!RFC7469] contains a JSON-based
+    definition for reporting individual pin validation failures.
+
+  * The Domain-based Message Authentication, Reporting, and Conformance (DMARC)
+    [@!RFC7489] contains an XML-based reporting format for aggregate and
+    detailed email delivery errors.
+
+
+
+# Reporting Policy
+
+SMTP TLSRPT policies are distributed via DNS from the Policy Domain's zone,
+either through a new resource record, or as TXT records (similar to DMARC
+policies) under the name `_smtp_tlsrpt`. (Current implementations deploy as TXT
+records.) For example, for the Policy Domain `example.com`, the recipient's
+SMTP STS policy can be retrieved from `_smtp_tlsrpt.example.com`.
+
+(Future implementations may move to alternate methods of policy discovery or
+distribution. See the section _Future_ _Work_ for more discussion.)
+
+Policies consist of the following directives:
+
+   * `version`: This value MUST be equal to `TLSRPT1`.
+   
+   * `aggregate-report-uri`: A URI specifying the endpoint to which aggregate
+     information about policy failures should be sent (see the section
+     _Reporting Schema_ for more information). Two URI schemes are supported:
+     `mailto` and `https`.
+
+	 * In the case of `https`, reports should be submitted via POST
+     ([@!RFC2818]) to the specified URI.
```

Comment:
Do we want t specify anything around https validation? Should validation happen at all?